### PR TITLE
feat(pair): Create various UX for pair/unsupported

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* This page is confusing and not ideal for our users and we want to render various new UX to help them complete the pair flow

This commit:
* Adjusts 'Pair using an app' screen according to design, only show users this when the URL reflects a state that the user tried pairing with their system camera
* Adds a new 'Connecting your mobile device' state for when mobile users (not through system camera) reach this page, and conditionally renders an alert with 'Download Firefox now' link for iOS and Android if user is not on FF
* Previously, the 'default state' for this page was the system camera screen. This commit adds a new generic error state
* Tweaks pair/failure to match the new style/image, hides "Sign in with email" text if button is not shown
* Adds automatic Glean click events cad_redirect_mobile_download and cad_redirect_mobile_learn_more, adds custom events cad_redirect_mobile_view and cad_redirect_desktop_default_view

closes FXA-10136

--

These screenshots have been sent to Eduardo and are approved from what he saw. You can see these by changing the UA in dev tools.

Edit: these are mostly up to date, but now I've added a class so `firefox.com/pair` is always on the same line.

image
image
image
image

bonus `pair/failure` updates:
image
image

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache [hunk 1]
```diff
@@ -1,12 +1,9 @@
 <div class="card-base">
-  <header>
-    {{#isDesktopNonFirefox}}
+  {{#showCADHeader}}
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache:2]: This conditional was easier than juggling it here in the mustache file.

File: packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache [hunk 2]
```diff
@@ -25,11 +22,35 @@
...
+      {{#isSystemCameraUrl}}
+        <div class="bg-image-pair-fail"></div>
+          <h2 id="pair-unsupported-header" class="card-header" tabindex="-1">{{#t}}Pair using an app{{/t}}</h1>
+        <p class="mt-4">{{#t}}Did you use the system camera? You must pair from within a Firefox app.{{/t}}</p>
+      {{/isSystemCameraUrl}}
+      {{^isSystemCameraUrl}}
+        {{^isFirefox}}
+          <div class="flex bg-orange-50 px-4 py-3 rounded-lg mb-8">
+            <div class="bg-icon-warning bg-contain bg-center bg-no-repeat w-10" aria-label="{{#t}}Attention:{{/t}}" role="img"></div>
+            <p class="ps-3 text-sm">{{#t}}Oops! It looks like you’re not using Firefox.{{/t}} <a href="{{{ escapedMobileDownloadLink }}}" target="_blank" class="link-blue" glean-data-id="cad_redirect_mobile_download">{{#t}}Download Firefox now{{/t}}</a></p>
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/templates/pair/unsupported.mustache:35]: I _think_ I'm probably supposed to use `unsafeTranslate` and embed the link? But these also felt okay separate.

File: packages/fxa-content-server/app/scripts/templates/pair/failure.mustache [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
     <div class="error"></div>
     <div class="success"></div>
 
     <form novalidate>
       <div class="bg-image-pair-fail"></div>
 
-      <p>{{#t}}The setup process was terminated. Please sign in with your email.{{/t}}</p>
+      <header>
+        <h1 id="fxa-pair-failure-header" class="card-header">
+          {{#t}}Device pairing failed.{{/t}}
```
Comment (LZoog) [inline packages/fxa-content-server/app/scripts/templates/pair/failure.mustache:11]: I'll remove the period on this with my next push.

Also just noticed, `The setup couldn't be completed.` should use a curly quote so I'll get that too.

File: packages/fxa-content-server/app/scripts/templates/pair/failure.mustache [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
       <div class="bg-image-pair-fail"></div>
 
-      <p>{{#t}}The setup process was terminated. Please sign in with your email.{{/t}}</p>
+      <header>
+        <h1 id="fxa-pair-failure-header" class="card-header">
+          {{#t}}Device pairing failed{{/t}}
+        </h1>
+      </header>
+
+      <p class="mt-3">{{#t}}The setup couldn’t be completed.{{/t}}{{#showSigninLink}} {{#t}}Please sign in with your email.{{/t}}{{/showSigninLink}}</p>
```
Comment (vbudhram) [inline packages/fxa-content-server/app/scripts/templates/pair/failure.mustache:15]: QA recently reopened https://mozilla-hub.atlassian.net/browse/FXA-10258 and I have been stratching my head on this. I think the best thing to do is to remove the button completely and tell the user to login with their email.

File: packages/fxa-content-server/app/scripts/views/pair/unsupported.js [hunk 5]
```diff
@@ -22,21 +23,38 @@ class PairUnsupportedView extends FormView {
 
   setInitialContext(context) {
     const uap = this.getUserAgent();
+    const isFirefox = uap.isFirefox();
+    const isMobile = uap.isAndroid() || uap.isIos();
     // Assume the user is on non-Firefox desktop in this case.
-    const isDesktopNonFirefox =
-      !uap.isFirefox() && !uap.isAndroid() && !uap.isIos();
+    const isDesktopNonFirefox = !isFirefox && !isMobile;
+    const isSystemCameraUrl = this.getHashParams(['channel_id']).channel_id;
```
Comment (vbudhram) [inline packages/fxa-content-server/app/scripts/views/pair/unsupported.js:30]: Not sure if this is the best way to flag this. The `channel_id` is on both desktop and mobile pages. Mobile does have a `channel_key` param that might be used instead.

Also on mobile, I think the we might be able to check that the user has opened in Safari and no Firefox iOS using user agents.

File: packages/fxa-content-server/app/scripts/templates/pair/failure.mustache (hunk 4 above)
Comment (vbudhram) [reply packages/fxa-content-server/app/scripts/templates/pair/failure.mustache:15]: We should also be able to close the above ticket as well.

**Review by vbudhram — APPROVED** [review]
@LZoog I didn't test all of the states but overall seems to work well. Lmk your thoughts on the questions.

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-content-server/app/scripts/views/pair/unsupported.js (hunk 5 above)
Comment (LZoog) [reply packages/fxa-content-server/app/scripts/views/pair/unsupported.js:30]: Thanks for calling this out! Originally I was checking for `channel_id` and `channel_key`, but I couldn't seem to get the `channel_key` check to work when `channel_id` was present for some reason and thought we wouldn't see either on desktop. Is checking for these two values sufficient, or should we check for both values plus `isMobile`?

File: packages/fxa-content-server/app/scripts/templates/pair/failure.mustache (hunk 4 above)
Comment (LZoog) [reply packages/fxa-content-server/app/scripts/templates/pair/failure.mustache:15]: Hmm was that a requested feature? If we shouldn't spend more time trying to fix it (I'm also not sure what's going on without investigating, I see the params are passed), I'm OK with removing it and revisiting when we do React work on that page.

File: packages/fxa-content-server/app/scripts/views/pair/unsupported.js (hunk 5 above)
Comment (vbudhram) [reply packages/fxa-content-server/app/scripts/views/pair/unsupported.js:30]: I think it is fine to check for all the conditions (channel_id, channel_key, isMobile). I think we still might have to tweak things but the messaging is generic ish enough that it should be ok.

File: packages/fxa-content-server/app/scripts/templates/pair/failure.mustache (hunk 4 above)
Comment (LZoog) [reply packages/fxa-content-server/app/scripts/templates/pair/failure.mustache:15]: Chatted in Slack and Vijay will remove this with the linked ticket in a quick follow up after this lands.

**Comment by LZoog** [comment]
Going to go ahead and merge, test is an unrelated MySQL timeout.

**Pull request merged**